vm-validator/src/vm_validator.rs

```diff
@@ -91,18 +261,17 @@ impl VMValidator {
 
     fn restart(&mut self) -> Result<()> {
         let db_state_view = self.db_state_view();
-
-        self.state_view = db_state_view.clone().into();
-        self.vm = Self::new_vm_for_validation(&self.state_view);
-        self.module_storage =
-            CachedDbStateView::from(db_state_view).into_aptos_code_storage(self.vm.environment());
-
+        self.state.reset_all(db_state_view.into());
         Ok(())
     }
 
     fn notify_commit(&mut self) {
         let db_state_view = self.db_state_view();
-        self.state_view = db_state_view.into();
+
+        // On commit, we need to update the state view, but not the configs or module caches. If
+        // configs change, validator will be restarted, and for module caches we ensure that read
+        // modules are up-to-date.
```

Comment:
> If configs change, validator will be restarted

this comment is a bit odd, as AptosVM contains onchain configs as well, right?
if so - this is safe I assume because epoch change creates a new one, not because of restart?

Reply:
Updated the comment